<launch>
    <!-- If a node config file argument is provided by the caller we will load it instead of the example config -->
    <arg name="node_config_file" default="$(find rosbag_cloud_recorders)/config/rolling_recorder_sample_configuration.yaml" />

    <!-- The length of each bag file in seconds.
         Acceptable values: positive integers
         If not given defaults to 30 seconds
         If an invalid value is given the node will exit and log an error message -->
    <arg name="bag_rollover_time" default="30" />

    <!-- The length of time recordings are kept. Any rosbag older than this time will be deleted.
         Acceptable values: positive integers
         If not given defaults to 300 seconds
         If an invalid value is given the node will exit and log an error message -->
    <arg name="max_record_time" default="300" />


    <!-- Directory where rosbags will be written.
         Defaults to ~/.ros/rosbag_uploader/ -->
    <arg name="write_directory" default="$(env HOME)/.ros/rosbag_uploader/" />

    <!-- The topics that should be recorded into the rosbag.
         Acceptable values: any list of strings or an empty list.
         e.g. '/topic1 /topic2 /topic3'
         If list is empty or no value is given defaults to all topics.
         If a topic in the list doesn't exist when recording starts
         no warning will be given. Behavior here depends on ROSBag implementaiton. -->
    <arg name="topics_to_record" default='-a' />

    <node name="rosbag_record" pkg="rosbag" type="record" args='--split
                                                                --duration $(arg bag_rollover_time)
                                                                --output-prefix $(arg write_directory)
                                                                $(arg topics_to_record)'>
          <param name="rosbag_record_mkdir" command="mkdir -p $(arg write_directory)" />
    </node>

    <node name="rolling_recorder" pkg="rosbag_cloud_recorders" type="rolling_recorder" >
      <rosparam command="load" file="$(arg node_config_file)" />
      <param name="bag_rollover_time" value="$(arg bag_rollover_time)" />
      <param name="max_record_time" value="$(arg max_record_time)" />
      <param name="write_directory" value="$(arg write_directory)" />
    </node>

</launch>
